### ISSUE TITLE
0.7.0 does not install with pypy 3.10 or 3.11

### ISSUE THREAD

**Issue by zx80** [mypyc/librt#21]
`librt` version _0.7.0_ does not seem to compile with pypy 3.10 or 3.11:

Should it declare and/or check the incompatibility somewhere instead of failing downstream while compiling a file?

```
      building 'librt.internal' extension
      creating build/temp.linux-x86_64-pypy310
      gcc -pthread -DNDEBUG -O2 -fPIC -I. -I/home/runner/work/flask-simple-auth/flask-simple-auth/venv/include -I/opt/hostedtoolcache/PyPy/3.10.16/x64/include/pypy3.10 -c exc_ops.c -o build/temp.linux-x86_64-pypy310/exc_ops.o -O3
      In file included from CPy.h:12,
                       from exc_ops.c:8:
      pythonsupport.h:38:1: warning: data definition has no type or storage class
         38 | _Py_IDENTIFIER(__mro_entries__);
            | ^~~~~~~~~~~~~~
      pythonsupport.h:38:1: warning: type defaults to ‘int’ in declaration of ‘_Py_IDENTIFIER’ [-Wimplicit-int]
      pythonsupport.h:38:1: warning: parameter names (without types) in function declaration
      pythonsupport.h: In function ‘update_bases’:
      pythonsupport.h:60:50: error: ‘PyId___mro_entries__’ undeclared (first use in this function)
         60 |         if (PyObject_GetOptionalAttrString(base, PyId___mro_entries__.string, &meth) < 0) {
            |                                                  ^~~~~~~~~~~~~~~~~~~~
      pythonsupport.h:60:50: note: each undeclared identifier is reported only once for each function it appears in
      pythonsupport.h: At top level:
      pythonsupport.h:113:1: warning: data definition has no type or storage class
        113 | _Py_IDENTIFIER(__init_subclass__);
            | ^~~~~~~~~~~~~~
      pythonsupport.h:113:1: warning: type defaults to ‘int’ in declaration of ‘_Py_IDENTIFIER’ [-Wimplicit-int]
      pythonsupport.h:113:1: warning: parameter names (without types) in function declaration
      pythonsupport.h: In function ‘init_subclass’:
      pythonsupport.h:125:12: warning: implicit declaration of function ‘_PyObject_GetAttrId’; did you mean ‘PyObject_GetAttr’? [-Wimplicit-function-declaration]
        125 |     func = _PyObject_GetAttrId(super, &PyId___init_subclass__);
            |            ^~~~~~~~~~~~~~~~~~~
            |            PyObject_GetAttr
      pythonsupport.h:125:40: error: ‘PyId___init_subclass__’ undeclared (first use in this function)
        125 |     func = _PyObject_GetAttrId(super, &PyId___init_subclass__);
            |                                        ^~~~~~~~~~~~~~~~~~~~~~
```

**Comment by ilevkivskyi** [comment]
Yes, PyPy is unfortunately not supported yet, see README and discussion in https://github.com/mypyc/librt/issues/16 for workarounds.

Is there a way to explicitly declare that we don't support PyPy?

**Comment by zx80** [comment]
> Is there a way to explicitly declare that we don't support PyPy?

Probably in  `pyproject.toml`, see [there](https://packaging.python.org/en/latest/specifications/dependency-specifiers/). I'm unclear where it can be declared in the module, but it sure can be declared when the dependency is used by another module, say `mypy`.
